group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -8641,6 +8665,130 @@ public void testListGroups() {
         assertEquals(expectAllGroupMap, actualAllGroupMap);
     }
 
+    @Test
+    public void testDescribeGroupStableForDynamicMember() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        JoinGroupResponseData responseData = context.joinGenericGroupAsDynamicMemberAndCompleteRebalance("group-id");
```

Comment:
The weirdness comes from the fact that joinGenericGroupAsDynamicMemberAndCompleteRebalance does not replay the generated records so the lastWrittenOffset is not updated and a new snapshot is not created. I tried to add the logic to replay them but that breaks a few tests because they assume that the GenericGroup object is updated but it is replaced when records are replayed. We should fix this but this is for another PR. Instead of doing this, we could just replay the relevant records to build the state that we want. Example:

```
# Create first group.
context.replay(newGroupMetadataRecord(....));
# Commit it.
context.commit()
# Create second group.
context.replay(newGroupMetadataRecord(....));

# Verify that first group is returned and second one is returned as dead

# Commit

# Verify that both groups are returned
```

What do you think?